src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -296,34 +297,33 @@ private function fixPhpDoc(Tokens $tokens, int $index, array &$uses, string $nam
 
         $phpDoc = $tokens[$index];
         $phpDocContent = $phpDoc->getContent();
-        Preg::matchAll('#@([^\s]+)(\s+)([a-zA-Z0-9_\\\\]+)#', $phpDocContent, $matches);
+        $phpDocContentNew = Preg::replaceCallback('/(\*\s*@)([^\s]+)(\s+)('.TypeExpression::REGEX_TYPES.')(?!\S)/', function ($matches) use ($allowedTags, &$uses, $namespaceName) {
+            if (!\in_array($matches[2], $allowedTags, true)) {
+                return $matches[0];
+            }
 
-        if ([] !== $matches[0]) {
-            foreach ($matches[3] as $i => $typeName) {
-                if (!\in_array($matches[1][$i], $allowedTags, true)) {
-                    continue;
-                }
+            // TODO fix all, even complex/nested, phpdoc types
```

Comment:
leftover, or actual awaiting TODO?

Reply:
TODO for the future

Reply:
please use `@TODO` then, define what is "all" etc.
when I read it, i have no clue what is this about.

Reply:
done

Reply:
`done` only partially ;)

2nd part is not yet addressed
> define what is "all" etc.
> when I read it, i have no clue what is this about.

Reply:
please help me with wording - by "all" I mean "all", ie. to walk TypeExpression and fix all non-reserved types

Reply:
done